sync-pipelines/src/commonMain/kotlin/com/quran/shared/pipeline/SyncEnginePipeline.kt

```diff
@@ -0,0 +1,123 @@
+package com.quran.shared.pipeline
+
+import co.touchlab.kermit.Logger
+import com.quran.shared.mutations.LocalModelMutation
+import com.quran.shared.mutations.RemoteModelMutation
+import com.quran.shared.persistence.repository.PageBookmarksSynchronizationRepository
+import com.quran.shared.syncengine.AuthenticationDataFetcher
+import com.quran.shared.syncengine.LocalModificationDateFetcher
+import com.quran.shared.syncengine.LocalMutationsFetcher
+import com.quran.shared.syncengine.PageBookmark
+import com.quran.shared.syncengine.PageBookmarksSynchronizationConfigurations
+import com.quran.shared.syncengine.ResultNotifier
+import com.quran.shared.syncengine.SynchronizationClient
+import com.quran.shared.syncengine.SynchronizationClientBuilder
+import com.quran.shared.syncengine.SynchronizationEnvironment
+
+interface SyncEngineCallback {
```

Comment:
I don't think we should hav this - we should update the `SyncEnginePipeline` to have a method that returns a `Flow<SyncState>` - that "Something" then is what people who listen to / consume this flow handle.

so we can do something like:

```kotlin
sealed class SyncState {
   data class Error(exception: Exception): SyncState()
   data class Success(
        val newRemoteMutations: List<RemoteModelMutation<PageBookmark>>,
        val processedLocalMutations: List<LocalModelMutation<PageBookmark>>
   ): SyncState()
}
```